test/any/confluence/set_ops-test.es6.js

```diff
@@ -206,4 +231,158 @@ describe('Set ops', () => {
           .then(done.fail, done);
     });
   });
+
+  describe('Intersect', () => {
+    it('should yield full set on INTERSECT(sameSet)', done => {
+      var set = MDAO.create({of: Num});
+      var sameSet = MDAO.create({of: Num});
+
+      for (var i = 0; i < 10; i++) {
+        set.put(mkNum(i));
+        sameSet.put(mkNum(i));
+      }
+
+      set.orderBy(Num.ID).select(E.INTERSECT(sameSet))
+          .then(sink => {
+            expect(sink.a.map(num => num.id))
+                .toEqual([0, 1, 2, 3, 4, 5, 6, 7, 8, 9]);
+            done();
+          });
+    });
+
+    it('should yield null set on INTERSECT(nullSet)', done => {
+      var set = MDAO.create({of: Num});
+      var nullSet = MDAO.create({of: Num});
+
+      for (var i = 0; i < 10; i++) {
+        set.put(mkNum(i));
+      }
+
+      set.orderBy(Num.ID).select(E.INTERSECT(nullSet))
+          .then(sink => {
+            expect(sink.a.map(num => num.id)).toEqual([]);
+            done();
+          });
+    });
+
+    it('should remove some items when secondary is subset', done => {
+      var set = MDAO.create({of: Num});
+      var secondary = MDAO.create({of: Num});
+
+      for (var i = 0; i < 10; i++) {
+        set.put(mkNum(i));
+        if ((i % 2) === 0) {
+          secondary.put(mkNum(i));
+        }
+      }
+
+      // {0..9} INTERSECT {0, 2, 4, 6, 8}
+      set.orderBy(Num.ID).select(E.INTERSECT(secondary))
+          .then(sink => {
+            expect(sink.a.map(num => num.id))
+                .toEqual([0, 2, 4, 6, 8]);
+
+            done();
+          });
+    });
+
+    it('should remove some items when primary is subset', done => {
```

Comment:
Exclude might be a better word than remove, since if you are thinking about it in terms of the mutation applied to primary, the intersect operation is not doing anything.